apollo-federation/src/operation/mod.rs

```diff
@@ -2820,27 +2821,28 @@ impl SelectionSet {
         schema: &ValidFederationSchema,
         option: NormalizeSelectionOption,
     ) -> Result<SelectionSet, FederationError> {
-        let mut normalized_selection_map = SelectionMap::new();
+        let mut normalized_selections = Self {
+            schema: schema.clone(),
+            type_position: parent_type.clone(),
```

Comment:
good catch! this was done in the JS `lazyMap` itself and not the normalize method